lib/mocha/expectation.rb

```diff
@@ -265,6 +265,8 @@ def with_no_block_given
     #
     # May be called multiple times on the same expectation for consecutive invocations.
     #
+    # If no +parameters+ are given then it will yield the block provided.
```

Comment:
This appears incorrect to me. This is just a special case of
> it yields with the specified +parameters+

from https://github.com/freerange/mocha/pull/496/files#diff-4c63dbab29ecbfb14e097947e3e0b295ad1382843978026979062d85b5287288R264

If we wanted to be more explicit about it, something like this would be more accurate
```suggestion
    # If no +parameters+ are given then it will yield to the block provided without any parameters.
```

This also makes me think, BTW, that perhaps there's a need to make the [introductory comment](https://github.com/freerange/mocha/pull/496/files#diff-4c63dbab29ecbfb14e097947e3e0b295ad1382843978026979062d85b5287288R264) clearer. Something like:

> Modifies expectation so that when the expected method is called, it yields with the specified +parameters+ to the block provided when calling. If no block is provided at the time of calling, the method will still yield, resulting in a +LocalJumpError+, as is standard ruby behavior.

Reply:
For comparison purposes, I'm copying the current version of the introductory documentation in here:

> Modifies expectation so that when the expected method is called, it yields with the specified +parameters+ (even if no block is provided, in which case yielding will result in a +LocalJumpError+).

Reply:
@nitishr I agree with both your suggested changes. Thanks.